src/libs/error.ts

```diff
@@ -13,18 +15,29 @@ export const reportError = async (title, obj) => {
   notify('error', title, { detail: await (obj instanceof Response ? obj.text() : obj) })
 }
 
+export type ErrorCallback = (error: unknown) => void | Promise<void>
+const withErrorHandlingFn = <R, F extends AnyPromiseFn>(
+  callback: ErrorCallback,
+  fn: GenericPromiseFn<F, R | void>
+): GenericPromiseFn<F, R | void> => async (...args: Parameters<F>): Promise<R | void> => {
+    try {
+      return await fn(...args)
+    } catch (error) {
+      await callback(error)
+    }
+  }
+
 /**
  * Invoke the `callback` with any error thrown when evaluating the async `fn` with `...args`.
  */
-export const withErrorHandling = _.curry((callback, fn) => async (...args) => {
-  try { return await fn(...args) } catch (error) { await callback(error) }
-})
+export const withErrorHandling = safeCurry(withErrorHandlingFn)
 
+// TODO type this properly
```

Comment:
Is there a ticket for the TODO? Should it be linked here?

Reply:
I can create one on UIE board and link here and in similar comments.

Reply:
I'll add this ticket link to the TODOs before merge:

https://broadworkbench.atlassian.net/browse/UIE-67